### DISCUSSION TITLE
Using single sample as all pitches of an instrument

### DISCUSSION THREAD

**Discussion by KontinuumLab** [pschatzmann/arduino-audio-tools#1317] [Q&A] [answered]
I am looking to make a hand-pan instrument using Arduino-STK and audio-tools on an ESP32.
I recently made a drum kit, with some help from Phil to get more samples into the STK-"Drummer". Works great! Thanks Phil!
I've been able to put some hand pan samples into the Drummer, with a separate sample for each pitch, to get a functioning prototype, but I had to reduce the duration of each sample considerably, to make everything fit in memory.
Now, I really want the full duration of the hand pan sound, which is part of its charm, so I'm hoping to be able to use a single sample, and then simply reproduce it at different speeds to get the different pitches. Hopefully with up to 4 note polyphony.
**Is there STK-functionality for reproducing several instances of the same sample at a time at different speeds?** 
I've been messing around with MemoryStream, but so far have gotten nowhere.
Any help would be greatly appreciated.
Thanks!

**Comment by pschatzmann** [comment, marked as answer]
I don't know stk good enough to provide an answer for stk. 

But using the AudioTools you could resample or us some pitch shifting: Both is explained in the Wiki
